tests/stages/arxiv/test_arxiv_source.py

```diff
@@ -0,0 +1,220 @@
+
+    def mock_load():
+        nonlocal call_count
+        call_count += 1
+        if call_count < 5:
+            raise pypdf.errors.PdfStreamError()
+
+        if call_count == 5:
+            return ["unittest"]
+
+        assert False, "Should not be called more than 5 times"
+
+    mock_pdf_loader.return_value = mock_pdf_loader
+    mock_pdf_loader.load.side_effect = mock_load
+    stage = ArxivSource(config, query="unittest", cache_dir="/does/not/exist")
+    documents = stage._process_pages('/does/not/exist/fake.pdf')
+
+    assert stage._total_pages == 1
+    assert call_count == 5
+
+    assert documents == ["unittest"]
+
+
+@pytest.mark.parametrize("chunk_size", [200, 1000])
+def test_splitting_pages(config: Config,
+                         pdf_file: str,
+                         langchain: types.ModuleType,
+                         chunk_size: int,
+                         dataset_cudf: DatasetManager):
+    chunk_overlap = 100
+    content = "Morpheus\nunittest"
+    while len(content) < chunk_size:
+        content += f'\n{string.ascii_lowercase}'
+
+    # The test PDF is quite small, so we need to pad it out to get multiple pages
+    # I checked this won't evently divide into either of the two values for chunk_size
+    if len(content) != chunk_size:
+        split_point = content.rfind('\n')
+        first_row = content[0:split_point]
+        second_row = content[split_point + 1:]
+        while len(second_row) < chunk_overlap:
+            split_point = content[0:split_point].rfind('\n')
+            second_row = content[split_point + 1:]
+
+        page_content_col = [first_row, second_row]
+    else:
+        page_content_col = [content]
+
+    num_expected_chunks = len(page_content_col)
+    source_col = []
+    page_col = []
+    for _ in range(num_expected_chunks):
+        source_col.append(pdf_file)
+        page_col.append(0)
+
+    expected_df = cudf.DataFrame({"page_content": page_content_col, "source": source_col, "page": page_col})
+
+    loader = langchain.document_loaders.PyPDFLoader(pdf_file)
+    documents = loader.load()
+    assert len(documents) == 1
+
+    document = documents[0]
+    document.page_content = content
+
+    stage = ArxivSource(config, query="unittest", chunk_size=chunk_size, chunk_overlap=chunk_overlap)
+    msg = stage._splitting_pages(documents)
+
+    assert stage._total_chunks == num_expected_chunks
+    dataset_cudf.assert_compare_df(msg.df, expected_df)
+
+
+def test_splitting_pages_no_chunks(config: Config,
+                                   pdf_file: str,
+                                   langchain: types.ModuleType,
+                                   dataset_cudf: DatasetManager):
+    content = "Morpheus\nunittest"
+    page_content_col = [content]
+    source_col = [pdf_file]
+    page_col = [0]
+
+    expected_df = cudf.DataFrame({"page_content": page_content_col, "source": source_col, "page": page_col})
+
+    loader = langchain.document_loaders.PyPDFLoader(pdf_file)
```

Comment:
Same issue here; is there a specific reason we need to test langchain's document loaders?

Reply:
Same, its the required input for the `_splitting_pages` method.